dali/operators/sequence/optical_flow/optical_flow_impl/optical_flow_impl.cc

```diff
@@ -85,22 +116,28 @@ OpticalFlowImpl::OpticalFlowImpl(dali::optical_flow::OpticalFlowParams params, s
 
   auto grid_sizes = GetCapabilities(of_inst_.nvOFGetCaps, of_handle_,
                                     NV_OF_CAPS_SUPPORTED_HINT_GRID_SIZES);
-  IsGridSupported(grid_sizes, out_grid_size_, "Output");
 
   auto hint_sizes = GetCapabilities(of_inst_.nvOFGetCaps, of_handle_,
                                     NV_OF_CAPS_SUPPORTED_HINT_GRID_SIZES);
-  IsGridSupported(hint_sizes, hint_grid_size_, "Hint");
 
   width_min_ = GetCapabilities(of_inst_.nvOFGetCaps, of_handle_, NV_OF_CAPS_WIDTH_MIN);
   height_min_ = GetCapabilities(of_inst_.nvOFGetCaps, of_handle_, NV_OF_CAPS_HEIGHT_MIN);
   width_max_ = GetCapabilities(of_inst_.nvOFGetCaps, of_handle_, NV_OF_CAPS_WIDTH_MAX);
   height_max_ = GetCapabilities(of_inst_.nvOFGetCaps, of_handle_, NV_OF_CAPS_HEIGHT_MAX);
+
+  if (width_min_.size()) default_init_params_.width = width_min_[0];
```

Comment:
What happens if we raise an exception between successful CreateOf in `L115` and those lines 128-130 adjusting the `default_init_params_`? We have `of_handle_` but `!is_initialized_`, so `default_init_params_` will be used in the destructor, right? Is this intended, is this OK?

Reply:
Then `nvOFInit` fail, `nvOFDestroy` fail as well inside `DestroyOf`, and it should result in `std::terminate();`. At this point, fatal crash is probably the best we can do. As in normal conditions, no matter what configuration the user provided the code between L115 and L128 should not fail (if OF is available we should get any capabilities).